### ISSUE TITLE
Invalid var_sources causes build to error without any hint

### ISSUE THREAD

**Issue by evanchaoli** [concourse/concourse#7927]
## Summary

If a pipeline contains an invalid `var_sources`, then build will immediately fail without any error message. As users cannot get hint of why build fails, they may have to ask support from OPs, which would significantly increase OPs cost.

## Steps to reproduce

Submit the following pipeline to a cluster:

```yaml
var_sources:
- name: vs
  type: vault
  config:
    url: http://fake-vault.com
    path_prefix: some
    client_token: ((my-vault-token))

jobs:
- name: job
  plan:
  - task: task1
    config:
      platform: linux
      image_resource:
        type: mock
        source: {mirror_self: true}
      run:
        path: echo
        args: [((vs:arg))]
```

Then trigger a build of the job, the job will immediately fail without any hint message:

![build-error-no-log](https://user-images.githubusercontent.com/18585861/148322565-8cc0cbf7-2e3c-422f-9e4d-5dae8f45b8ae.png)

From ATC logs, I can find an error:

```
web_1     | {"timestamp":"2022-01-06T03:18:54.401787200Z","level":"error","source":"atc","message":"atc.tracker.notify.run.failed-to-create-run-state","data":{"build":"3","build_id":34,"error":"evaluate var_source 'vs' error: undefined vars: my-vault-token","job":"job","pipeline":"tiv","session":"26.7.2","team":"main"}}
web_1     | {"timestamp":"2022-01-06T03:18:54.412388000Z","level":"info","source":"atc","message":"atc.tracker.notify.run.finish.errored","data":{"build":"3","build_id":34,"error":"evaluate var_source 'vs' error: undefined vars: my-vault-token","job":"job","pipeline":"tiv","session":"26.7.2.3","team":"main"}}
```

## Expected results

If an error is caused by pipeline definition, users should be able to view the error, so that users get idea of how to fix the error.

## Actual results

Users cannot see any error.

## Triaging info

* Concourse version: 7.6.0/master branch
* Browser (if applicable): NA
* Did this used to work?

**Comment by evanchaoli** [comment]
Actually, the error message is available in UI, just need to display the error message in some manner.

For example, for such a bad build, `plan` API will return:

```
{"schema":"exec.v2","plan":{"id":"61dbed25","do":[{"id":"61dbed24","task":{"name":"task1","privileged":false}}]}}
```

And `events` API will return:

```
{"data":{"message":"evaluate var_source 'vs' error: undefined vars: my-vault-token","origin":{"id":"61dbed25"},"time":1641803073},"event":"error","version":"4.1","event_id":"1"}
```

Here `origin`'s id "61dbed25" is the plan id. So if build errors, and there is an event with origin id of plan id, then display the message.

Unfortunately, I'm not a front-end guy. Anyone can help with this issue will be greatly appreciated.

**Comment by clarafu** [comment]
Looks like when vars are undefined, it also does not show up in the web UI. For example, I have an undefined var that is found in a resource source configuration

```
name: 1m
type: time
source:
  interval: 1m
  unique: ((param))
```

In the resource page I do not see any errors 
![image](https://user-images.githubusercontent.com/19290703/149373888-47b8adac-8007-4c9f-b759-7a85c5f9d6f2.png)

But in the web logs I see the errors `atc.tracker-imb.run.finish.errored : resource config creds evaluation: undefined vars: param`

**Comment by marco-m-pix4d** [comment]
@aliculPix4D is this the same error you mentioned a few times?

**Comment by aliculPix4D** [comment]
@marco-m-pix4d Yes, I think the issue that I opened: https://github.com/concourse/concourse/issues/8350 is solved with this [PR](https://github.com/concourse/concourse/pull/8015).

I will just asked the authors (@xtremerui, @clarafu, @evanchaoli ) to confirm, after which I will close my issue.

**Comment by evanchaoli** [comment]
7.8.0 is out today, this issue should have been resolved with 7.8.0.
